gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java

```diff
@@ -273,6 +270,15 @@ protected Script produceScript(final String traversalSource, final Bytecode o) {
                         script.append("int32(");
                         convertToScript(arguments[i]);
                         script.append(")");
+                    } else if ((methodName.equals("inject") ||methodName.equals("withSack")  ||
+                                methodName.equals("sample") || methodName.equals("with") || methodName.equals("constant"))
```

Comment:
I would recommend defining a static array/container of these values, and using a single contains method to check for it.

Reply:
Made an edit for it.